### ISSUE TITLE
go get honnef.co/go/tools/cmd/staticcheck failing

### ISSUE THREAD

**Issue by nelsonpina** [dominikh/go-tools#531]
CI pipeline for my Go services is breaking because it can't get staticcheck

tested in both:
- go version go1.11.11 linux/amd64
- go version go1.12.2 darwin/amd64

```
» go get honnef.co/go/tools/cmd/staticcheck
go: finding honnef.co/go/tools/cmd/staticcheck latest
go: finding honnef.co/go/tools/cmd latest
go: finding honnef.co/go/tools v0.0.0-2019.2.1
go: downloading honnef.co/go/tools v0.0.0-2019.2.1
go: extracting honnef.co/go/tools v0.0.0-2019.2.1
go: finding github.com/rogpeppe/go-internal v1.3.0
go: finding github.com/google/renameio v0.1.0
go: finding golang.org/x/tools v0.0.0-20190621195816-6e04913cbbac
go: finding golang.org/x/mod v0.0.0-20190513183733-4bf6d317e70e
go: finding golang.org/x/crypto v0.0.0-20190510104115-cbcb75029529
go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: finding golang.org/x/sys v0.0.0-20190412213103-97732733099d
go: finding gopkg.in/errgo.v2 v2.1.0
go: finding cloud.google.com/go v0.37.3
go: finding cloud.google.com/go v0.37.3-alpha
go: finding cloud.google.com/go v0.37.2
go: finding cloud.google.com/go v0.37.1
go: finding cloud.google.com/go v0.37.0
go: finding github.com/googleapis/gax-go/v2 v2.0.3
go: finding cloud.google.com/go v0.36.0
go: finding cloud.google.com/go v0.35.1
go: finding cloud.google.com/go v0.35.0
go: finding github.com/googleapis/gax-go/v2 v2.0.2
go: finding github.com/googleapis/gax-go/v2 v2.0.1
go: finding github.com/googleapis/gax-go/v2 v2.0.0
go: github.com/googleapis/gax-go/v2@v2.0.0: missing github.com/googleapis/gax-go/go.mod and .../v2/go.mod at revision v2.0.0
panic: error loading module requirements

goroutine 1 [running]:
cmd/go/internal/mvs.Downgrade.func2(0xc00011a1c0, 0x1f, 0xc000265c54, 0x6)
        /usr/local/Cellar/go/1.12.2/libexec/src/cmd/go/internal/mvs/mvs.go:317 +0x676
cmd/go/internal/mvs.Downgrade(0xc00021dd50, 0x6, 0x0, 0x0, 0x16c8300, 0xc0001ca310, 0xc0003fe1a0, 0x1, 0x1, 0x10160df, ...)
        /usr/local/Cellar/go/1.12.2/libexec/src/cmd/go/internal/mvs/mvs.go:349 +0x5c7
cmd/go/internal/modget.runGet(0x1a48c60, 0xc000020050, 0x1, 0x1)
        /usr/local/Cellar/go/1.12.2/libexec/src/cmd/go/internal/modget/get.go:446 +0x20d3
main.main()
        /usr/local/Cellar/go/1.12.2/libexec/src/cmd/go/main.go:219 +0x7a4

```

**Comment by ainar-g** [comment]
Dup of #528.

**Comment by dominikh** [comment]
Closing as duplicate, and as something we can't really fix. It looks to be a bug in Go :-/

**Comment by nelsonpina** [comment]
yes, for now, the solution is to pinpoint version 2019.1

**Comment by dmitshur** [comment]
> It looks to be a bug in Go :-/

It was a bug in Go (specifically, issue golang/go#31942). It is fixed in the upcoming Go 1.13 release.

I can reproduce with Go 1.12.6 when starting with the following non-empty `go.mod` file:

```
$ cat go.mod 
module m

go 1.12

require (
	github.com/golang/protobuf v1.3.1 // indirect
	github.com/google/renameio v0.1.0 // indirect
	github.com/googleapis/gax-go/v2 v2.0.5 // indirect
	github.com/kisielk/gotool v1.0.0 // indirect
	github.com/rogpeppe/go-internal v1.3.0 // indirect
	golang.org/x/mod v0.0.0-20190513183733-4bf6d317e70e // indirect
	golang.org/x/net v0.0.0-20190628185345-da137c7871d7 // indirect
	golang.org/x/tools v0.0.0-20190621195816-6e04913cbbac // indirect
)
$ go get honnef.co/go/tools/cmd/staticcheck@2019.2
go: finding honnef.co/go/tools/cmd/staticcheck 2019.2
go: finding honnef.co/go/tools/cmd 2019.2
go: finding honnef.co/go/tools 2019.2
go: github.com/googleapis/gax-go/v2@v2.0.0: missing github.com/googleapis/gax-go/go.mod and .../v2/go.mod at revision v2.0.0
panic: error loading module requirements

goroutine 1 [running]:
cmd/go/internal/mvs.Downgrade.func2(0xc000115180, 0x1f, 0xc00012a754, 0x6)
	/usr/local/go/src/cmd/go/internal/mvs/mvs.go:317 +0x676
cmd/go/internal/mvs.Downgrade(0x1a151cd, 0x1, 0x0, 0x0, 0x16c85e0, 0xc00017c150, 0xc000216380, 0x1, 0x1, 0x400, ...)
	/usr/local/go/src/cmd/go/internal/mvs/mvs.go:349 +0x5c7
cmd/go/internal/modget.runGet(0x1a45ba0, 0xc000022050, 0x1, 0x1)
	/usr/local/go/src/cmd/go/internal/modget/get.go:446 +0x20d3
main.main()
	/usr/local/go/src/cmd/go/main.go:219 +0x7a4
```

It's fixed when using `go1.13beta1`:

```
$ go1.13beta1 get honnef.co/go/tools/cmd/staticcheck@2019.2
go: finding honnef.co/go/tools/cmd/staticcheck 2019.2
go: finding honnef.co/go/tools/cmd 2019.2
go: finding honnef.co/go/tools 2019.2
go: downloading honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099
go: extracting honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099
```

**Comment by nelsonpina** [comment]
@dmitshur 
```
the easiest workaround for now would be to install staticcheck using an empty (or non-existent) go.mod file
```
thanks for pointing that out. yes, that allows the installation of the latest version, but for my CI jobs it's cleaner (and a better practice) to pinpoint a version than having to create an empty .mod to install the latest version of `staticcheck`
